src/components/pages/donation_form/DonationSummary.vue

```diff
@@ -9,13 +9,13 @@
 		<div class="switcher">
 			<div>
 				<DonorSummary
-					v-if="hasAddressSummary"
+					v-if="address"
```

Comment:
I'd see this type handling with `undefined` as a decrease in readability.

Ideally I would wish for a local, computed property `addressIsComplete`/`hasAddress` in this component that knows how to calculate/evaluate the state of the `address` property.

Reply:
> I'd see this type handling with `undefined` as a decrease in readability.
> 
> Ideally I would wish for a local, computed property `addressIsComplete`/`hasAddress` in this component that knows how to calculate/evaluate the state of the `address` property.

@moiikana I also had the same feelings. But Gabriel made me aware of `undefined` being a legit type : https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html. And now I feel more comfortable with this PR.

Reply:
I don't think we're losing that much readability in the larger context. Eg if you look at the `hasAddressSummary` computed item you need to parse a long evaulation, and also then know what an Address contains. IMO it's easier to think about as `if an address is defined` than if an address has `firstName`, `lastName`, `companyName`, `street`, `zip` etc.

Plus it reduces maintenance and modularises things a bit, and if we adopt this pattern everywhere then it'll be consistent and we'll get used to it